python/lsst/meas/extensions/ngmix/processCoaddsNGMix.py

```diff
@@ -109,53 +112,58 @@ def cdict(self):
             self._cdict = self.config.toDict()
         return self._cdict
 
-    def run(self, images, ref, replacers, imageId, butler=None, kwargs_butler=None):
+    def run(self, catexps: List[CatalogExposure], cat_ref: SourceCatalog):
```

Comment:
Are these changes for gen3?

Reply:
Yes, it's also part of the `MultibandFit` interface. You can't pass `butler`s into base `Task`s in gen3*; all I/O is handled in the `PipelineTask`. I don't think `noiseReplacer` is a persisted dataset type, so I added a metadata dict to `CatalogExposure` to add whatever you want to each catalog-exposure pair.

*well, it's not enforced yet, and I don't think I removed everything from the various `Task`s that is invalid in gen3 - this is a first pass attempt to maintain compatibility with gen2 while it's still possible.